web/src/components/courtlist/CourtListTableActionBarGroup.vue

```diff
@@ -43,35 +66,129 @@
+  import { BinderService } from '@/services';
+  import { useCommonStore } from '@/stores';
 
   const props = defineProps<{
     selected: CourtListAppearance[];
   }>();
 
+  const binderService = inject<BinderService>('binderService');
+  const commonStore = useCommonStore();
+
+  if (!binderService) {
+    throw new Error('Service is undefined.');
+  }
+
   const emit = defineEmits<{
     (e: 'view-case-details', appearances: CourtListAppearance[]): void;
     (e: 'view-key-documents', appearances: CourtListAppearance[]): void;
+    (e: 'unique-civil-file-selected', appearance: CourtListAppearance): void;
+    (e: 'view-judicial-binders', appearances: CourtListAppearance[]): void;
   }>();
 
+  const previousCivilFileIds = new Set<string>();
+  const binderCounts = ref<Record<string, number>>({});
+  const binderRequests = ref<Promise<any>[]>([]);
+  const binderLoading = computed(() => binderRequests.value.length > 0);
+
   const groupedSelections = computed(() => {
     const groups: Record<string, CourtListAppearance[]> = {};
+    const currentCivilFileIds = new Set<string>();
+    
     for (const item of props.selected) {
       const group = getCourtClassLabel(item.courtClassCd);
       if (!groups[group]) {
         groups[group] = [];
       }
       groups[group].push(item);
+
+      // Track civil/family appearances by physicalFileId
+      if (!isCourtClassLabelCriminal(group) && item.physicalFileId) {
+        currentCivilFileIds.add(item.physicalFileId);
+        
+        // Emit event if this is a new unique physicalFileId
+        if (!previousCivilFileIds.has(item.physicalFileId)) {
+          emit('unique-civil-file-selected', item);
+        }
+      }
     }
 
+    // Update the previous set for next comparison
+    previousCivilFileIds.clear();
+    currentCivilFileIds.forEach(id => previousCivilFileIds.add(id));
+
     return groups;
   });
 
+  // Watch for changes in selected civil/family files and fetch binder counts
+  watch(
+    () => props.selected,
+    async (newSelected, oldSelected) => {
+      const getCivilFiles = (items: CourtListAppearance[]) =>
+        items.filter((item) => 
+          !isCourtClassLabelCriminal(getCourtClassLabel(item.courtClassCd)) && 
+          item.physicalFileId
+        );
+
+      const civilFiles = getCivilFiles(newSelected);
+      const isSelection = newSelected.length > (oldSelected?.length || 0);
+
+      if (civilFiles.length === 0) {
+        binderCounts.value = {};
+        return;
+      }
+      if(!isSelection) {
+        // No need to fetch if all files are already tracked
+        const currentFileIds = new Set(civilFiles.map((f) => f.physicalFileId));
+        binderCounts.value = Object.fromEntries(
+          Object.entries(binderCounts.value).filter(([fileId]) => currentFileIds.has(fileId))
+        );
+        return;
+      }
+      // Fetch binder count for newly selected file
+      const fileId = newSelected[newSelected.length - 1].physicalFileId;
```

Comment:
the seems to work for adding one new entry as long as that new entry goes to the end of the array - should this be more generic or could that wait for requirements?